django/core/handlers/asgi.py

```diff
@@ -186,9 +186,13 @@ async def handle(self, scope, receive, send):
         if request is None:
             body_file.close()
             await self.send_response(error_response, send)
+            await sync_to_async(error_response.close, thread_sensitive=True)()
```

Comment:
Is this safe to remove? 🤔 (We `return` here, so when is `error_response.close` called?)

Reply:
Interesting question - I'm not sure, my only reason to add it was that this is code was in send_response, and therefore it would have been called before this change. While I don't know of any reason why the django code needs it, the `request_finished` signal is documented so applications can use it, and I guess could in theory depend on this behaviour?

Reply:
If it's me, it's a separate ticket. 

* Is it an issue?
* If so, can we demonstrate that in a test. 

But either way, it's not the issue here is it?

Reply:
Ok, I think the conclusion we are coming to is this is not needed - there's no test cases that currently fail and its not obvious that it is needed. I can remove.

Reply:
This PR is intended for backporting, so we don't want to make any unrelated changes in it. Please revert, and feel-free to send a follow up PR when this one is merged.

Reply:
Reviewing this conversation now, I'm confused by my initial comment. 

I think this **is** needed 

@jthorniley's: 

> ...reason to add it was that this is code was in send_response, and therefore it would have been called before this change.

That sounds right. 

Sorry if I confused the discussion there. (I can't recall exactly what I thought at the time.)